OPHD/UI/FactoryProduction.cpp

```diff
@@ -212,19 +212,25 @@ void FactoryProduction::update()
 
 	Window::update();
 
-	const int labelWidth = 120;
-	auto position = mRect.startPoint() + NAS2D::Vector{constants::MARGIN * 2 + static_cast<int>(mProductGrid.width()), 25};
-	drawLabelAndValueRightJustify(position, labelWidth, "Turns Completed:", std::to_string(mFactory->productionTurnsCompleted()) + " of " + std::to_string(mProductCost.turnsToBuild()));
+	StringTable stringTable(2, 5);
+	stringTable.setPosition({ mRect.x() + constants::MARGIN * 2 + static_cast<int>(mProductGrid.width()), mRect.y() + 25 });
```

Comment:
I wanted to remove the static_cast on this line, but it is only tangentially related to StringTable. Is there any reason to keep this cast in place, it seemed not necessary to me.

Reply:
Try it and see.

----

Often it's necessary for type conversion in cases like this. The temporary being passed as the parameter is initialized with `{}`, which does not allow narrowing conversions (of non-literals). That means the types in the expressions have to match exactly to be able to construct an object with the correct `BaseType`.

The constructor for `NAS2D::Vector<BaseType>` is defined to take two `BaseType` parameters. For the `{}` brace initialization to be valid, the types of the expressions need to match exactly. If one expression is of type `int`, while the other is of type `float`, it will be an error. The `BaseType` can not be correctly deduced in that case. Sometimes you need to add in a value that's a different type, and so you need to convert it to prevent it from changing the type of the expression. In particular, the `float` type tends to dominate, so you can end up with expressions of the type: `float + int ==> float`.

In this case, it appears the target type is `float`, so casting to `int` does indeed seem a bit odd here. You're probably right to remove it.

----

The other possibility, is that a lot of the code in this project will cast to `int` as a way to force rounding of a value. What the code really means in such case is `std::floor(value)`. If we find that is the case, and the rounding is needed, replacing the cast with `std::floor` makes more sense when you want to round a `float` type and end up with a final expression that's still of `float` type. The might be the intention here.

Reply:
OK, I'll try removing it. Agreed that std::floor would indicate intent better than a static_cast in these situations.

I have found with experience that while you can draw sprites on screen using floats, the image isn't as sharp when fractional positions are provided.

-Brett